ports/esp32s2/common-hal/pwmio/PWMOut.c

```diff
@@ -128,7 +128,13 @@ pwmout_result_t common_hal_pwmio_pwmout_construct(pwmio_pwmout_obj_t *self,
     if (ledc_channel_config(&(self->chan_handle))) {
         return PWMOUT_INITIALIZATION_ERROR;
     }
-
+    
+    // check the frequency -- 0 is not valid
+    // maybe also want to reject frequency > system_clock / 2 ??
```

Comment:
Should this check for frequency too high?
I don't know the ESP32-S2 platform well enough to decide.

Reply:
> Should this check for frequency too high?

No, that is not needed as the idf functions do a check for correct `duty_cycle` and `frequency` combinations.